JavadocTagContinuationIndentation: doesn't report violations for html elements

https://checkstyle.org/config_javadoc.html#JavadocTagContinuationIndentation
1. Test.java:
```
gauravpunjabi@Gauravs-MacBook-Pro:~/Coding/Open Source/checkstyle$ javac Test.java 
gauravpunjabi@Gauravs-MacBook-Pro:~/Coding/Open Source/checkstyle$ cat -n Test.java 
     1  public class Test {
     2          /**
     3           * Some description here.
     4           *
     5           * @param a
     6           * Text description here. // violation
     7           * @param b
     8           * <b>Some description in bold</b> // OK
     9           */
    10          public void test(int a, int b) {}
    11  }
```
2. Config file:
```
gauravpunjabi@Gauravs-MacBook-Pro:~/Coding/Open Source/checkstyle$ cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <property name="charset" value="ASCII" />
    <property name="severity" value="error" />
    <property name="fileExtensions" value="java" />

    <module name="TreeWalker">
        <module name="JavadocTagContinuationIndentation" />
    </module>
</module>
```

3. Output:
```
gauravpunjabi@Gauravs-MacBook-Pro:~/Coding/Open Source/checkstyle$ java -jar target/checkstyle-8.32-SNAPSHOT-all.jar -c config.xml Test.java 
Starting audit...
[ERROR] /Users/gauravpunjabi/Coding/Open Source/checkstyle/Test.java:6: Line continuation have incorrect indentation level, expected level should be 4. [JavadocTagContinuationIndentation]
Audit done.
Checkstyle ends with 1 errors.
```

#### Expected
Line 8 should be reported as a violation, it is not correctly indented.

#### Explaination
JavadocTagContinutation does not report indentation violations for `HTML_ELEMENT`. As discussed in here https://github.com/checkstyle/checkstyle/pull/8002#discussion_r414565394 .
`HTML_ELEMENT` must also be properly intended to make the docs look good and easily readable.

I should be able to handle it under the scope of #8177, the root cause is similar to #5711.

```
D:\checkstyletest>type config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="severity" value="error"/>
    <module name="TreeWalker">
        <module name="JavadocTagContinuationIndentation">
          <property name="offset" value="4"/>
        </module>

    </module>
</module>

D:\checkstyletest>type test\Test.java
public class Test {
          /**
           * Some description here.
           *
           * @param a (text is needed here until #8002 is merged)
           * Text description here. // violation
           * @param b (text is needed here until #8002 is merged)
           * <b>Some description in bold</b> // violation
           */
          public void test(int a, int b) {}
}
D:\checkstyletest>java -jar -Duser.language=en -Duser.country=US checkstyle-8.32-SNAPSHOT-all.jar -c config.xml test\Test.java
Starting audit...
[ERROR] D:\checkstyletest\test\Test.java:6: Line continuation have incorrect indentation level, expected level should be 4. [JavadocTagContinuationIndentation]
[ERROR] D:\checkstyletest\test\Test.java:8: Line continuation have incorrect indentation level, expected level should be 4. [JavadocTagContinuationIndentation]
Audit done.
```

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTagContinuationIndentationCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTagContinuationIndentationCheck.java
@@ -175,16 +175,44 @@ public class JavadocTagContinuationIndentationCheck extends AbstractJavadocCheck
             final List<DetailNode> textNodes = getAllNewlineNodes(ast);
             for (DetailNode newlineNode : textNodes) {
                 final DetailNode textNode = JavadocUtil.getNextSibling(newlineNode);
-                if (textNode.getType() == JavadocTokenTypes.TEXT) {
-                    final String text = textNode.getText();
-                    if (!CommonUtil.isBlank(text.trim())
-                            && (text.length() <= offset
-                                    || !text.substring(1, offset + 1).trim().isEmpty())) {
-                        log(textNode.getLineNumber(), MSG_KEY, offset);
-                    }
+                if (textNode.getType() == JavadocTokenTypes.TEXT && isViolation(textNode)) {
+                    log(textNode.getLineNumber(), MSG_KEY, offset);
                 }
             }
         }
+    }
+
+    /**
+     * Checks if a text node meets the criteria for a violation.
+     * If the text is shorter than {@code offset} characters, then a violation is
+     * detected if the text is not blank or the next node is not a newline.
+     * If the text is longer than {@code offset} characters, then a violation is
+     * detected if any of the first {@code offset} characters are not blank.
+     *
+     * @param textNode the node to check.
+     * @return true if the node has a violation.
+     */
+    private boolean isViolation(DetailNode textNode) {
+        boolean result = false;
+        final String text = textNode.getText();
+        if (text.length() <= offset) {
+            if (CommonUtil.isBlank(text)) {
+                final DetailNode nextNode = JavadocUtil.getNextSibling(textNode);
+                if (nextNode != null && nextNode.getType() != JavadocTokenTypes.NEWLINE) {
+                    // text is blank but line hasn't ended yet
+                    result = true;
+                }
+            }
+            else {
+                // text is not blank
+                result = true;
+            }
+        }
+        else if (!CommonUtil.isBlank(text.substring(1, offset + 1))) {
+            // first offset number of characters are not blank
+            result = true;
+        }
+        return result;
     }
 
     /**

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTagContinuationIndentationCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTagContinuationIndentationCheckTest.java
@@ -109,4 +109,26 @@ public class JavadocTagContinuationIndentationCheckTest
                 expected);
     }
 
+    @Test
+    public void testBlockTag() throws Exception {
+        final DefaultConfiguration checkConfig =
+                createModuleConfig(JavadocTagContinuationIndentationCheck.class);
+        checkConfig.addAttribute("offset", "4");
+        final String[] expected = {
+            "17: " + getCheckMessage(MSG_KEY, 4),
+            "28: " + getCheckMessage(MSG_KEY, 4),
+            "38: " + getCheckMessage(MSG_KEY, 4),
+            "58: " + getCheckMessage(MSG_KEY, 4),
+            "60: " + getCheckMessage(MSG_KEY, 4),
+            "70: " + getCheckMessage(MSG_KEY, 4),
+            "71: " + getCheckMessage(MSG_KEY, 4),
+            "72: " + getCheckMessage(MSG_KEY, 4),
+            "82: " + getCheckMessage(MSG_KEY, 4),
+            "83: " + getCheckMessage(MSG_KEY, 4),
+            "84: " + getCheckMessage(MSG_KEY, 4),
+        };
+        verify(checkConfig, getPath("InputJavadocTagContinuationIndentationBlockTag.java"),
+                expected);
+    }
+
 }

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadoctagcontinuationindentation/InputJavadocTagContinuationIndentationBlockTag.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadoctagcontinuationindentation/InputJavadocTagContinuationIndentationBlockTag.java
@@ -1,0 +1,102 @@
+package com.puppycrawl.tools.checkstyle.checks.javadoc.javadoctagcontinuationindentation;
+
+import java.io.Serializable;
+
+/**
+ * Config:
+ * offset = 4
+ */
+public class InputJavadocTagContinuationIndentationBlockTag {
+
+    /**
+     * Example from issue 5711.
+     * Returns the value represented by the specified string of the specified
+     * type. Returns 0 for types other than float, double, int, and long.
+     * @param text the string to be parsed.
+     * @param type the token type of the text. Should be a constant of
+     * {@link TokenTypes}. // violation
+     * @return the double value represented by the string argument.
+     */
+    public static double parseDouble(String text, int type) {
+        return 0;
+    }
+
+    /**
+     * Javadoc.
+     *
+     * @param x this line is normal
+     * {@code this} line is wrongly indented // violation
+     */
+    public void newlineThenBlockTag(int x) {
+        // do stuff
+    }
+
+    /**
+     * Not enough indentation.
+     *
+     * @param x this line is normal
+     *   {@code this} line is wrongly indented // violation
+     */
+    public void partialIndent(int x) {
+        // do stuff
+    }
+
+    /**
+     * There can be a newline but nothing follows it.
+     *
+     * @param x input
+     * @return itself
+     * */ // ok
+    public int identity(int x) {
+        return x;
+    }
+
+    /**
+     * Javadoc.
+     *
+     * @param args // ok
+     * {@code this} line is not correctly indented // violation
+     *     {@code this} // ok
+     * <pre>this line is not correctly indented</pre> // violation
+     */
+    public void multipleLines1(String args) {
+        // do stuff
+    }
+
+    /**
+     * Javadoc.
+     *
+     * @return false always // ok
+     * {@code this} line is not correctly indented // violation
+     * {@code this} line is not correctly indented // violation
+     * <pre>this line is not correctly indented</pre> // violation
+     */
+    public boolean isMultipleLines2() {
+        return false;
+    }
+
+    /**
+     * Javadoc from regression test case (apache-ant).
+     *
+     * @param c the command line which will be configured
+     * if the commandline is initially null, the function is a noop
+     * otherwise the function append to the commandline arguments concerning
+     * <ul> // violation, not correctly indented
+     * <li> // ok, inside a HTML element
+     * cvs package // ok, inside a HTML element
+     * </li> // ok, inside a HTML element
+     * <li> // ok, inside a HTML element
+     * compression // ok, inside a HTML element
+     * </li> // ok, inside a HTML element
+     * <li> // ok, inside a HTML element
+     * quiet or reallyquiet // ok, inside a HTML element
+     * </li> // ok, inside a HTML element
+     * <li>cvsroot</li> // ok, inside a HTML element
+     * <li>noexec</li> // ok, inside a HTML element
+     * </ul> // ok, inside a HTML element
+     */
+    public String regressionNestedHtml(CharSequence c) {
+        return "";
+    }
+
+}

